pycuda doesn't build with cuda 13

**Describe the bug**
Build crashes with latest cuda 13

**To Reproduce**
pip install pycuda

**Expected behavior**
not crash

**Environment (please complete the following information):**
 - OS: ubuntu 24.04
 - CUDA version: 13.0
 - CUDA driver version: latest 580.65.06    
  - PyCUDA version: latest 2025.1.1
  - Python version: 3.10 & 3.12

**Additional context**
Add any other context about the problem here.

We've got this on Arch Linux with CUDA 13 and GCC 15:

```
src/cpp/cuda.hpp:862:54: error: invalid conversion from ‘unsigned int’ to ‘CUctxCreateParams*’ {aka ‘CUctxCreateParams_st*’} [-fpermissive]
  862 |     CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
      |                                                      ^~~~~
      |                                                      |
      |                                                      unsigned int
src/cpp/cuda.hpp:132:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED’
  132 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
src/cpp/cuda.hpp:862:47: error: too few arguments to function ‘CUresult cuCtxCreate_v4(CUctx_st**, CUctxCreateParams*, unsigned int, CUdevice)’
  862 |     CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
src/cpp/cuda.hpp:132:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED’
  132 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
/opt/cuda/include/cuda.h:90:45: note: declared here
   90 | #define cuCtxCreate                         cuCtxCreate_v4
      |                                             ^~~~~~~~~~~~~~
/opt/cuda/include/cuda.h:6275:18: note: in expansion of macro ‘cuCtxCreate’
 6275 | CUresult CUDAAPI cuCtxCreate(CUcontext *pctx, CUctxCreateParams *ctxCreateParams, unsigned int flags, CUdevice dev);
      |                  ^~~~~~~~~~~
src/cpp/cuda.hpp: In constructor ‘pycuda::texture_reference::texture_reference()’:
```

```
 7173 | __CUDA_DEPRECATED CUresult CUDAAPI cuCtxDetach(CUcontext ctx);
      |                                    ^~~~~~~~~~~
src/cpp/cuda.hpp: In static member function ‘static CUsharedconfig pycuda::context::get_shared_config()’:
src/cpp/cuda.hpp:821:54: warning: ‘CUresult cuCtxGetSharedMemConfig(CUsharedconfig*)’ is deprecated [-Wdeprecated-declarations]
  821 |         CUDAPP_CALL_GUARDED(cuCtxGetSharedMemConfig, (&config));
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:7218:36: note: declared here
 7218 | __CUDA_DEPRECATED CUresult CUDAAPI cuCtxGetSharedMemConfig(CUsharedconfig *pConfig);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In static member function ‘static void pycuda::context::set_shared_config(CUsharedconfig)’:
src/cpp/cuda.hpp:827:54: warning: ‘CUresult cuCtxSetSharedMemConfig(CUsharedconfig)’ is deprecated [-Wdeprecated-declarations]
  827 |         CUDAPP_CALL_GUARDED(cuCtxSetSharedMemConfig, (config));
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:7273:36: note: declared here
 7273 | __CUDA_DEPRECATED CUresult CUDAAPI cuCtxSetSharedMemConfig(CUsharedconfig config);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘pycudaboost::shared_ptr<pycuda::context> pycuda::device::make_context(unsigned int)’:
src/cpp/cuda.hpp:862:54: error: invalid conversion from ‘unsigned int’ to ‘CUctxCreateParams*’ {aka ‘CUctxCreateParams_st*’} [-fpermissive]
  862 |     CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
      |                                                      ^~~~~
      |                                                      |
      |                                                      unsigned int
src/cpp/cuda.hpp:132:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED’
  132 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
src/cpp/cuda.hpp:862:47: error: too few arguments to function ‘CUresult cuCtxCreate_v4(CUctx_st**, CUctxCreateParams*, unsigned int, CUdevice)’
  862 |     CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
src/cpp/cuda.hpp:132:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED’
  132 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
/usr/local/cuda/include/cuda.h:90:45: note: declared here
   90 | #define cuCtxCreate                         cuCtxCreate_v4
      |                                             ^~~~~~~~~~~~~~
/usr/local/cuda/include/cuda.h:6275:18: note: in expansion of macro ‘cuCtxCreate’
 6275 | CUresult CUDAAPI cuCtxCreate(CUcontext *pctx, CUctxCreateParams *ctxCreateParams, unsigned int flags, CUdevice dev);
      |                  ^~~~~~~~~~~
src/cpp/cuda.hpp: In constructor ‘pycuda::texture_reference::texture_reference()’:
src/cpp/cuda.hpp:1130:45: warning: ‘CUresult cuTexRefCreate(CUtexref_st**)’ is deprecated [-Wdeprecated-declarations]
 1130 |       { CUDAPP_CALL_GUARDED(cuTexRefCreate, (&m_texref)); }
      |                             ~~~~~~~~~~~~~~~~^~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22724:36: note: declared here
22724 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefCreate(CUtexref *pTexRef);
      |                                    ^~~~~~~~~~~~~~
src/cpp/cuda.hpp: In destructor ‘pycuda::texture_reference::~texture_reference()’:
src/cpp/cuda.hpp:1140:56: warning: ‘CUresult cuTexRefDestroy(CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1140 |           CUDAPP_CALL_GUARDED_CLEANUP(cuTexRefDestroy, (m_texref));
      |                                       ~~~~~~~~~~~~~~~~~^~~~~~~~~~
src/cpp/cuda.hpp:152:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_CLEANUP’
  152 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22744:36: note: declared here
22744 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefDestroy(CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::texture_reference::set_array(pycudaboost::shared_ptr<pycuda::array>)’:
src/cpp/cuda.hpp:1152:47: warning: ‘CUresult cuTexRefSetArray(CUtexref, CUarray, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1152 |         CUDAPP_CALL_GUARDED(cuTexRefSetArray, (m_texref,
      |                             ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
 1153 |             ary->handle(), CU_TRSA_OVERRIDE_FORMAT));
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:21935:36: note: declared here
21935 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetArray(CUtexref hTexRef, CUarray hArray, unsigned int Flags);
      |                                    ^~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘pycuda::pycuda_size_t pycuda::texture_reference::set_address(CUdeviceptr, unsigned int, bool)’:
src/cpp/cuda.hpp:1160:49: warning: ‘CUresult cuTexRefSetAddress_v2(size_t*, CUtexref, CUdeviceptr, size_t)’ is deprecated [-Wdeprecated-declarations]
 1160 |         CUDAPP_CALL_GUARDED(cuTexRefSetAddress, (&byte_offset,
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:134:45: note: declared here
  134 | #define cuTexRefSetAddress                  cuTexRefSetAddress_v2
      |                                             ^~~~~~~~~~~~~~~~~~~~~
/usr/local/cuda/include/cuda.h:22011:36: note: in expansion of macro ‘cuTexRefSetAddress’
22011 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetAddress(size_t *ByteOffset, CUtexref hTexRef, CUdeviceptr dptr, size_t bytes);
      |                                    ^~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::texture_reference::set_address_2d(CUdeviceptr, const CUDA_ARRAY_DESCRIPTOR&, unsigned int)’:
src/cpp/cuda.hpp:1175:51: warning: ‘CUresult cuTexRefSetAddress2D_v3(CUtexref, const CUDA_ARRAY_DESCRIPTOR*, CUdeviceptr, size_t)’ is deprecated [-Wdeprecated-declarations]
 1175 |         CUDAPP_CALL_GUARDED(cuTexRefSetAddress2D, (m_texref, &descr, dptr, pitch));
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:142:45: note: declared here
  142 | #define cuTexRefSetAddress2D                cuTexRefSetAddress2D_v3
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/usr/local/cuda/include/cuda.h:22066:36: note: in expansion of macro ‘cuTexRefSetAddress2D’
22066 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetAddress2D(CUtexref hTexRef, const CUDA_ARRAY_DESCRIPTOR *desc, CUdeviceptr dptr, size_t Pitch);
      |                                    ^~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::texture_reference::set_format(CUarray_format, int)’:
src/cpp/cuda.hpp:1180:48: warning: ‘CUresult cuTexRefSetFormat(CUtexref, CUarray_format, int)’ is deprecated [-Wdeprecated-declarations]
 1180 |       { CUDAPP_CALL_GUARDED(cuTexRefSetFormat, (m_texref, fmt, num_packed_components)); }
      |                             ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22098:36: note: declared here
22098 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetFormat(CUtexref hTexRef, CUarray_format fmt, int NumPackedComponents);
      |                                    ^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::texture_reference::set_address_mode(int, CUaddress_mode)’:
src/cpp/cuda.hpp:1183:53: warning: ‘CUresult cuTexRefSetAddressMode(CUtexref, int, CUaddress_mode)’ is deprecated [-Wdeprecated-declarations]
 1183 |       { CUDAPP_CALL_GUARDED(cuTexRefSetAddressMode, (m_texref, dim, am)); }
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22141:36: note: declared here
22141 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetAddressMode(CUtexref hTexRef, int dim, CUaddress_mode am);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::texture_reference::set_filter_mode(CUfilter_mode)’:
src/cpp/cuda.hpp:1185:52: warning: ‘CUresult cuTexRefSetFilterMode(CUtexref, CUfilter_mode)’ is deprecated [-Wdeprecated-declarations]
 1185 |       { CUDAPP_CALL_GUARDED(cuTexRefSetFilterMode, (m_texref, fm)); }
      |                             ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22177:36: note: declared here
22177 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetFilterMode(CUtexref hTexRef, CUfilter_mode fm);
      |                                    ^~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::texture_reference::set_flags(unsigned int)’:
src/cpp/cuda.hpp:1188:47: warning: ‘CUresult cuTexRefSetFlags(CUtexref, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1188 |       { CUDAPP_CALL_GUARDED(cuTexRefSetFlags, (m_texref, flags)); }
      |                             ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22377:36: note: declared here
22377 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefSetFlags(CUtexref hTexRef, unsigned int Flags);
      |                                    ^~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘CUdeviceptr pycuda::texture_reference::get_address()’:
src/cpp/cuda.hpp:1193:49: warning: ‘CUresult cuTexRefGetAddress_v2(CUdeviceptr*, CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1193 |         CUDAPP_CALL_GUARDED(cuTexRefGetAddress, (&result, m_texref));
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:135:45: note: declared here
  135 | #define cuTexRefGetAddress                  cuTexRefGetAddress_v2
      |                                             ^~~~~~~~~~~~~~~~~~~~~
/usr/local/cuda/include/cuda.h:22404:36: note: in expansion of macro ‘cuTexRefGetAddress’
22404 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefGetAddress(CUdeviceptr *pdptr, CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘pycuda::array* pycuda::texture_reference::get_array()’:
src/cpp/cuda.hpp:1199:47: warning: ‘CUresult cuTexRefGetArray(CUarray_st**, CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1199 |         CUDAPP_CALL_GUARDED(cuTexRefGetArray, (&result, m_texref));
      |                             ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22431:36: note: declared here
22431 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefGetArray(CUarray *phArray, CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘CUaddress_mode pycuda::texture_reference::get_address_mode(int)’:
src/cpp/cuda.hpp:1205:53: warning: ‘CUresult cuTexRefGetAddressMode(CUaddress_mode*, CUtexref, int)’ is deprecated [-Wdeprecated-declarations]
 1205 |         CUDAPP_CALL_GUARDED(cuTexRefGetAddressMode, (&result, m_texref, dim));
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22486:36: note: declared here
22486 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefGetAddressMode(CUaddress_mode *pam, CUtexref hTexRef, int dim);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘CUfilter_mode pycuda::texture_reference::get_filter_mode()’:
src/cpp/cuda.hpp:1211:52: warning: ‘CUresult cuTexRefGetFilterMode(CUfilter_mode*, CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1211 |         CUDAPP_CALL_GUARDED(cuTexRefGetFilterMode, (&result, m_texref));
      |                             ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22512:36: note: declared here
22512 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefGetFilterMode(CUfilter_mode *pfm, CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘pycudaboost::python::tuple pycuda::texture_reference::get_format()’:
src/cpp/cuda.hpp:1220:48: warning: ‘CUresult cuTexRefGetFormat(CUarray_format*, int*, CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1220 |         CUDAPP_CALL_GUARDED(cuTexRefGetFormat, (&fmt, &num_channels, m_texref));
      |                             ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22540:36: note: declared here
22540 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefGetFormat(CUarray_format *pFormat, int *pNumChannels, CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘unsigned int pycuda::texture_reference::get_flags()’:
src/cpp/cuda.hpp:1228:47: warning: ‘CUresult cuTexRefGetFlags(unsigned int*, CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1228 |         CUDAPP_CALL_GUARDED(cuTexRefGetFlags, (&result, m_texref));
      |                             ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22699:36: note: declared here
22699 | __CUDA_DEPRECATED CUresult CUDAAPI cuTexRefGetFlags(unsigned int *pFlags, CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::surface_reference::set_array(pycudaboost::shared_ptr<pycuda::array>, unsigned int)’:
src/cpp/cuda.hpp:1261:48: warning: ‘CUresult cuSurfRefSetArray(CUsurfref, CUarray, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1261 |         CUDAPP_CALL_GUARDED(cuSurfRefSetArray, (m_surfref, ary->handle(), flags));
      |                             ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22787:36: note: declared here
22787 | __CUDA_DEPRECATED CUresult CUDAAPI cuSurfRefSetArray(CUsurfref hSurfRef, CUarray hArray, unsigned int Flags);
      |                                    ^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘pycuda::array* pycuda::surface_reference::get_array()’:
src/cpp/cuda.hpp:1268:48: warning: ‘CUresult cuSurfRefGetArray(CUarray_st**, CUsurfref)’ is deprecated [-Wdeprecated-declarations]
 1268 |         CUDAPP_CALL_GUARDED(cuSurfRefGetArray, (&result, m_surfref));
      |                             ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:22810:36: note: declared here
22810 | __CUDA_DEPRECATED CUresult CUDAAPI cuSurfRefGetArray(CUarray *phArray, CUsurfref hSurfRef);
      |                                    ^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In function ‘pycuda::texture_reference* pycuda::module_get_texref(pycudaboost::shared_ptr<module>, const char*)’:
src/cpp/cuda.hpp:1325:44: warning: ‘CUresult cuModuleGetTexRef(CUtexref_st**, CUmodule, const char*)’ is deprecated [-Wdeprecated-declarations]
 1325 |     CUDAPP_CALL_GUARDED(cuModuleGetTexRef, (&tr, mod->handle(), name));
      |                         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:7830:36: note: declared here
 7830 | __CUDA_DEPRECATED CUresult CUDAAPI cuModuleGetTexRef(CUtexref *pTexRef, CUmodule hmod, const char *name);
      |                                    ^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In function ‘pycuda::surface_reference* pycuda::module_get_surfref(pycudaboost::shared_ptr<module>, const char*)’:
src/cpp/cuda.hpp:1338:45: warning: ‘CUresult cuModuleGetSurfRef(CUsurfref_st**, CUmodule, const char*)’ is deprecated [-Wdeprecated-declarations]
 1338 |     CUDAPP_CALL_GUARDED(cuModuleGetSurfRef, (&sr, mod->handle(), name));
      |                         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:7864:36: note: declared here
 7864 | __CUDA_DEPRECATED CUresult CUDAAPI cuModuleGetSurfRef(CUsurfref *pSurfRef, CUmodule hmod, const char *name);
      |                                    ^~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::set_block_shape(int, int, int)’:
src/cpp/cuda.hpp:1363:34: warning: ‘CUresult cuFuncSetBlockShape(CUfunction, int, int, int)’ is deprecated [-Wdeprecated-declarations]
 1363 |             cuFuncSetBlockShape, (m_function, x, y, z), m_symbol);
      |             ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18177:36: note: declared here
18177 | __CUDA_DEPRECATED CUresult CUDAAPI cuFuncSetBlockShape(CUfunction hfunc, int x, int y, int z);
      |                                    ^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::set_shared_size(unsigned int)’:
src/cpp/cuda.hpp:1368:34: warning: ‘CUresult cuFuncSetSharedSize(CUfunction, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1368 |             cuFuncSetSharedSize, (m_function, bytes), m_symbol);
      |             ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18211:36: note: declared here
18211 | __CUDA_DEPRECATED CUresult CUDAAPI cuFuncSetSharedSize(CUfunction hfunc, unsigned int bytes);
      |                                    ^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::param_set_size(unsigned int)’:
src/cpp/cuda.hpp:1374:29: warning: ‘CUresult cuParamSetSize(CUfunction, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1374 |             cuParamSetSize, (m_function, bytes), m_symbol);
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18243:36: note: declared here
18243 | __CUDA_DEPRECATED CUresult CUDAAPI cuParamSetSize(CUfunction hfunc, unsigned int numbytes);
      |                                    ^~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::param_set(int, unsigned int)’:
src/cpp/cuda.hpp:1379:26: warning: ‘CUresult cuParamSeti(CUfunction, int, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1379 |             cuParamSeti, (m_function, offset, value), m_symbol);
      |             ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18276:36: note: declared here
18276 | __CUDA_DEPRECATED CUresult CUDAAPI cuParamSeti(CUfunction hfunc, int offset, unsigned int value);
      |                                    ^~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::param_set(int, float)’:
src/cpp/cuda.hpp:1384:24: warning: ‘CUresult cuParamSetf(CUfunction, int, float)’ is deprecated [-Wdeprecated-declarations]
 1384 |           cuParamSetf, (m_function, offset, value), m_symbol);
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18309:36: note: declared here
18309 | __CUDA_DEPRECATED CUresult CUDAAPI cuParamSetf(CUfunction hfunc, int offset, float value);
      |                                    ^~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::param_setv(int, void*, size_t)’:
src/cpp/cuda.hpp:1390:24: warning: ‘CUresult cuParamSetv(CUfunction, int, void*, unsigned int)’ is deprecated [-Wdeprecated-declarations]
 1390 |           cuParamSetv, (m_function, offset, buf, (unsigned int) len), m_symbol);
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18344:36: note: declared here
18344 | __CUDA_DEPRECATED CUresult CUDAAPI cuParamSetv(CUfunction hfunc, int offset, void *ptr, unsigned int numbytes);
      |                                    ^~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::param_set_texref(const pycuda::texture_reference&)’:
src/cpp/cuda.hpp:1394:63: warning: ‘CUresult cuParamSetTexRef(CUfunction, int, CUtexref)’ is deprecated [-Wdeprecated-declarations]
 1394 |         CUDAPP_CALL_GUARDED_WITH_TRACE_INFO(cuParamSetTexRef, (m_function,
      |                                             ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
 1395 |             CU_PARAM_TR_DEFAULT, tr.handle()), m_symbol);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                  
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18537:36: note: declared here
18537 | __CUDA_DEPRECATED CUresult CUDAAPI cuParamSetTexRef(CUfunction hfunc, int texunit, CUtexref hTexRef);
      |                                    ^~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::launch()’:
src/cpp/cuda.hpp:1401:23: warning: ‘CUresult cuLaunch(CUfunction)’ is deprecated [-Wdeprecated-declarations]
 1401 |             cuLaunch, (m_function), m_symbol);
      |             ~~~~~~~~~~^~~~~~~~~~~~
src/cpp/cuda.hpp:111:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED_WITH_TRACE_INFO’
  111 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
/usr/local/cuda/include/cuda.h:18396:36: note: declared here
18396 | __CUDA_DEPRECATED CUresult CUDAAPI cuLaunch(CUfunction f);
      |                                    ^~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::launch_grid(int, int)’:
src/cpp/cuda.hpp:1406:25: warning: ‘CUresult cuLaunchGrid(CUfunction, int, int)’ is deprecated [-Wdeprecated-declarations]
 1406 |           cuLaunchGrid, (m_function, grid_width, grid_height), m_symbol);
      |           ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:111:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED_WITH_TRACE_INFO’
  111 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
/usr/local/cuda/include/cuda.h:18450:36: note: declared here
18450 | __CUDA_DEPRECATED CUresult CUDAAPI cuLaunchGrid(CUfunction f, int grid_width, int grid_height);
      |                                    ^~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::launch_grid_async(int, int, const pycuda::stream&)’:
src/cpp/cuda.hpp:1411:32: warning: ‘CUresult cuLaunchGridAsync(CUfunction, int, int, CUstream)’ is deprecated [-Wdeprecated-declarations]
 1411 |             cuLaunchGridAsync, (m_function, grid_width, grid_height, s.handle()),
      |             ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:111:29: note: in definition of macro ‘CUDAPP_CALL_GUARDED_THREADED_WITH_TRACE_INFO’
  111 |       cu_status_code = NAME ARGLIST; \
      |                             ^~~~~~~
/usr/local/cuda/include/cuda.h:18512:36: note: declared here
18512 | __CUDA_DEPRECATED CUresult CUDAAPI cuLaunchGridAsync(CUfunction f, int grid_width, int grid_height, CUstream hStream);
      |                                    ^~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In member function ‘void pycuda::function::set_shared_config(CUsharedconfig)’:
src/cpp/cuda.hpp:1499:39: warning: ‘CUresult cuFuncSetSharedMemConfig(CUfunction, CUsharedconfig)’ is deprecated [-Wdeprecated-declarations]
 1499 |             cuFuncSetSharedMemConfig, (m_function, config), m_symbol);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:121:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED_WITH_TRACE_INFO’
  121 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
/usr/local/cuda/include/cuda.h:18591:36: note: declared here
18591 | __CUDA_DEPRECATED CUresult CUDAAPI cuFuncSetSharedMemConfig(CUfunction hfunc, CUsharedconfig config);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp: In function ‘void pycuda::initialize_profiler(const char*, const char*, CUoutput_mode)’:
src/cpp/cuda.hpp:2234:47: warning: ‘CUresult cuProfilerInitialize(const char*, const char*, CUoutput_mode)’ is deprecated [-Wdeprecated-declarations]
 2234 |     CUDAPP_CALL_GUARDED(cuProfilerInitialize, (config_file, output_file, output_mode));
      |                         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cpp/cuda.hpp:143:27: note: in definition of macro ‘CUDAPP_CALL_GUARDED’
  143 |     cu_status_code = NAME ARGLIST; \
      |                           ^~~~~~~
In file included from src/cpp/cuda.hpp:23:
/usr/local/cuda/include/cudaProfiler.h:143:36: note: declared here
  143 | __CUDA_DEPRECATED CUresult CUDAAPI cuProfilerInitialize(const char *configFile, const char *outputFile, CUoutput_mode outputMode);
      |                                    ^~~~~~~~~~~~~~~~~~~~
error: command '/usr/bin/aarch64-linux-gnu-g++' failed with exit code 1
The command '/bin/sh -c pip3 install pycuda || /tmp/build_pycuda.sh' returned a non-zero code: 1
[07:10:17] ===================================================================================== 
[07:10:17] ===================================================================================== 
[07:10:17] 💣 `jetson-containers build` failed after 150.0 seconds (2.5 minutes) 
[07:10:17] Error: Command 'DOCKER_BUILDKIT=0 docker build --network=host   --tag lerobot:r38.2.aarch64-cu130-24.04-pycuda   --file /home/ubuntu/Projects/jetson-containers/packages/cuda/pycuda/Dockerfile   --build-arg BASE_IMAGE=lerobot:r38.2.aarch64-cu130-24.04-cuda-python   --build-arg PYCUDA_VERSION="main"    /home/ubuntu/Projects/jetson-containers/packages/cuda/pycuda 2>&1 | tee /home/ubuntu/Projects/jetson-containers/logs/20250902_070747/build/31o34_lerobot_r38.2.aarch64-cu130-24.04-pycuda.txt; exit ${PIPESTATUS[0]}' returned non-zero exit status 1. 
[07:10:17] ===================================================================================== 
[07:10:17] ===================================================================================== 
[07:10:17] Failed building:  lerobot

Traceback (most recent call last):
  File "/home/ubuntu/Projects/jetson-containers/jetson_containers/build.py", line 129, in <module>
    build_container(**vars(args))
  File "/home/ubuntu/Projects/jetson-containers/jetson_containers/container.py", line 231, in build_container
    status = subprocess.run(cmd.replace(_NEWLINE_, ' '), executable='/bin/bash', shell=True, check=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'DOCKER_BUILDKIT=0 docker build --network=host   --tag lerobot:r38.2.aarch64-cu130-24.04-pycuda   --file /home/ubuntu/Projects/jetson-containers/packages/cuda/pycuda/Dockerfile   --build-arg BASE_IMAGE=lerobot:r38.2.aarch64-cu130-24.04-cuda-python   --build-arg PYCUDA_VERSION="main"    /home/ubuntu/Projects/jetson-containers/packages/cuda/pycuda 2>&1 | tee /home/ubuntu/Projects/jetson-containers/logs/20250902_070747/build/31o34_lerobot_r38.2.aarch64-cu130-24.04-pycuda.txt; exit ${PIPESTATUS[0]}' returned non-zero exit status 1.
 ```

--- a/src/cpp/cuda.hpp
+++ b/src/cpp/cuda.hpp
@@ -859,7 +859,13 @@ namespace pycuda
     context::prepare_context_switch();
 
     CUcontext ctx;
+
+#if CUDAPP_CUDA_VERSION >= 13000
+    CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, nullptr, flags, m_device));
+#else
     CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
+#endif
+
     boost::shared_ptr<context> result(new context(ctx));
     context_stack::get().push(result);
     return result;

